src/component/utils/exploration/__tests__/wrapBlockNodes.js

```diff
@@ -0,0 +1,580 @@
+  });
+
+  const contentState = ContentState.createFromBlockArray([
+    rootBlock,
+    new ContentBlockNode({
+      parent: 'A',
+      key: 'B',
+      text: 'first list item',
+      type: 'ordered-list-item',
+      children: List(['C']),
+    }),
+    new ContentBlockNode({
+      parent: 'B',
+      key: 'C',
+      text: 'header inside list',
+      type: 'header-one',
+    }),
+  ]);
+
+  const CustomBlock = props => {
+    return <div className={'custom-header-one'}>Custom Component</div>;
+  };
+
+  assertWrapBlockNodes({
+    ...PROPS,
+    contentState,
+    block: rootBlock,
+    blockRendererFn: block => {
+      if (block.getType() === 'header-one') {
+        return {
+          component: CustomBlock,
+        };
+      }
+      return null;
+    },
+    tree: BlockTree.generate(contentState, rootBlock, null),
+  });
+});
+
+test('renders block with nested children with custom component and editable prop', () => {
+  const rootBlock = new ContentBlockNode({
+    key: 'A',
+    text: '',
+    type: 'blockquote',
+    children: List(['B']),
+  });
+
+  const contentState = ContentState.createFromBlockArray([
+    rootBlock,
+    new ContentBlockNode({
+      parent: 'A',
+      key: 'B',
+      text: 'first list item',
+      type: 'ordered-list-item',
+      children: List(['C']),
+    }),
+    new ContentBlockNode({
+      parent: 'B',
+      key: 'C',
+      text: 'header inside list',
+      type: 'header-one',
+    }),
+  ]);
+
+  const CustomBlock = props => {
+    return (
+      <div className={'custom-header-one'}>
+        Custom Component {props.blockProps.foo}
+      </div>
+    );
+  };
+
+  assertWrapBlockNodes({
+    ...PROPS,
+    contentState,
+    block: rootBlock,
+    blockRendererFn: block => {
+      if (block.getType() === 'header-one') {
+        return {
+          component: CustomBlock,
+          editable: false,
+          props: {
+            foo: 'bar',
+          },
+        };
+      }
+      return null;
+    },
+    tree: BlockTree.generate(contentState, rootBlock, null),
+  });
+});
+
+test('renders block with nested children with blockStyleFn', () => {
+  const rootBlock = new ContentBlockNode({
+    key: 'A',
+    text: '',
+    type: 'blockquote',
+    children: List(['B']),
+  });
+
+  const contentState = ContentState.createFromBlockArray([
+    rootBlock,
+    new ContentBlockNode({
+      parent: 'A',
+      key: 'B',
+      text: 'first list item',
+      type: 'ordered-list-item',
+      children: List(['C']),
+    }),
+    new ContentBlockNode({
+      parent: 'B',
+      key: 'C',
+      text: 'header inside list',
+      type: 'header-one',
+    }),
+  ]);
+
+  assertWrapBlockNodes({
+    ...PROPS,
+    contentState,
+    block: rootBlock,
+    blockStyleFn: block => {
+      if (block.getType() === 'header-one') {
+        return 'My-fancy-custom-class';
+      }
+      return null;
+    },
+    tree: BlockTree.generate(contentState, rootBlock, null),
+  });
+});
```

Comment:
Really glad you added all these tests! ^_^

Would it make sense to add one with multiple sibling children at the same level? These are mostly [A -> [B -> [C]]], rather than [A -> [B, C]]
Like a List of children which all have type 'ordered-list-item' would, I expect, get wrapped in the same parent `ol`.